### DISCUSSION TITLE
How to use Powershell 7 instead of Powershell 5?

### DISCUSSION THREAD

**Discussion by InfospyPT** [microsoft/azure-devops-extension-tasks#486] [General]
Hello,

I have a task that I need to run under Powershell 7 instead of Powershell 5.

On the task.json file I have the following Execution definition:

"execution": {
    "PowerShell3": {
      "target": "task.ps1",
      "workingDirectory": "$(currentDirectory)",
      "argumentFormat": ""
    }

The task runs correctly on the Agent server but it runs on Powershell 5, how (if possible) can I run it on Powershell 7?

**Comment by jessehouwing** [comment]
No, the PowerShell handler uses PowerShell 5, unfortunately.

You can call `& pwsh script.ps1` from the entry point though... Or call PowerShell core from a node task. Check out the official PowerShell task to see how Microsoft does that. 

https://github.com/microsoft/azure-pipelines-tasks/blob/090322fa22fe2c8c165408f1fe555e8bddd82138/Tasks/PowerShellV2/powershell.ts

**Reply by InfospyPT** [reply]
If I understood correctly, it loads pwsh.exe (7.x) from powershell.exe(5.1) and then runs the script in pwsh.exe.

**Reply by jessehouwing** [reply]
Yeah and it has a fallback scenario where it runs pwsh.exe from Node on Linux and Mac.

**Comment by MadChipmunk03** [comment]
add *pwsh: true* to the inputs.

```yml
- task: PowerShell@2
  displayName:  RunWithPs7
  inputs:
    pwsh: true
```
    
source: https://microsoft-bitools.blogspot.com/2021/02/devops-snack-change-powershell-version.html
